### ISSUE TITLE
[React Native] – Error: Invalid response for blob

### ISSUE THREAD

**Issue by RafikiTiki** [matthieugicquel/shm#1]
Hey @matthieugicquel, first of all, thanks for creating this library. I'm currently evaluating various mocking projects for a React Native project. Besides mocking for the unit tests, I'm looking for a developer-friendly way to mock the API responses during E2E testing and development.

So far, I've tested the MSW but without success – I could not find a polyfill good enough to make it work in the React Native scenario. That's when I've stumbled upon your projects and decided to give it a try :)

Unfortunately, after following the quick and nice React Native guide in the readme file, I'm facing the following issue:

```
BUNDLE  ./index.js 

 LOG  Running "MockRepro" with {"rootTag":1,"initialProps":{}}
 ERROR  [Error: Invalid response for blob: {"name":"Test","description":"Test Description"}]
```

If I don't add any handlers to the mockServer (or add one that's not matching the request I'm firing) the error message is slightly different, but still the main culprit is the `Invalid response for blob`:

```
 BUNDLE  ./index.js 

 LOG  Running "MockRepro" with {"rootTag":11,"initialProps":{}}
 ERROR  [Error: Invalid response for blob: {"message":"No matching handler"}]
```

I tried using iOS simulator and a real device as well, with the same results. I'm using `react-query` as the networking library. I created [a minimal reproduction repo](https://github.com/RafikiTiki/shm-mock-repro) in case you'll find some time to help me with this issue. [Here's the SHM configuration](https://github.com/RafikiTiki/shm-mock-repro/blob/37b83e220cfd20ca0288558b7975c269f0f59c67/App.tsx#L66-L82 ).

Please let me know if you need anything from me :)

**Comment by matthieugicquel** [comment]
Hello @RafikiTiki, thanks for the detailed issue with the repro!

I found the cause: the react-native interceptor doesn't handle `blob` response types, but when using `fetch` (instead of eg `axios`), this is the response type that is used under the hood.

https://github.com/matthieugicquel/shm/blob/177d16ecfe2f51fa66434606e4087954cf51a11c/src/interceptors/RCTNetworking.ts#L18

🤦‍♂️🤦‍♂️

I'll work on fix (and make sure to test with `fetch` + `react-native` in the future, or introduce some E2E tests)

**Comment by RafikiTiki** [comment]
I'm glad I could help! Let me know if you'll need anything from me. Also, I'm happy to help with testing :)

**Comment by RafikiTiki** [comment]
BTW just checked, and replacing fetch with axios does the trick as a short-term solution i.e. replacing:

```ts
 const {isLoading, error, data} = useQuery('repoData', () =>
    fetch('http://example.com/test').then(res => res.json()),
  );
  ```

with

```ts
const {isLoading, error, data} = useQuery('repoData', () =>
    axios.get('http://example.com/test').then(res => res.data),
  );
  ```

does the trick 🚀

**Comment by matthieugicquel** [comment]
I've published v0.5.2 which should fix the issue. Feel free to reopen it if that's not the case 🙂
